pwndbg/dbg/lldb.py

```diff
@@ -221,6 +263,100 @@ def evaluate_expression(self, expression: str) -> pwndbg.dbg_mod.Value:
 
         return LLDBValue(value)
 
+    @override
+    def vmmap(self) -> pwndbg.dbg_mod.MemoryMap:
+        regions = self.process.GetMemoryRegions()
+
+        pages = []
+        for i in range(regions.GetSize()):
+            region = lldb.SBMemoryRegionInfo()
+            assert regions.GetMemoryRegionAtIndex(
+                i, region
+            ), "invalid region despite being in bounds"
+
+            perms = 0
+            if region.IsReadable():
+                perms |= os.R_OK
+            if region.IsWritable():
+                perms |= os.W_OK
+            if region.IsExecutable():
+                perms |= os.X_OK
+
+            # LLDB doesn't actually tell us the offset of the mapped file, just
+            # whether it is mapped or not.
+            offset = 0 if not region.IsMapped() else 1
```

Comment:
Is it ok we set offset=1 if its mapped?

Reply:
No, that's a mistake on my part. I totally forgot that that got changed to zero in a later commit, and didn't get backported. Lemme change it now.